src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -101,16 +102,55 @@ internal CloudTable GetDiagnosticEventsTable(DateTime? now = null)
             return _diagnosticEventsTable;
         }
 
-        private static string GetCurrentTableName(DateTime now)
+        private static string GetTableName(DateTime date)
         {
-            return $"{TableNamePrefix}{now:yyyyMM}";
+            return $"{TableNamePrefix}{date:yyyyMM}";
         }
 
         protected internal virtual async void OnFlushLogs(object state)
         {
             await FlushLogs();
         }
 
+        // TODO:// Call this ~1 minute after host startup to avoid blocking host start & impacting cold start.
+        private async void PurgePreviousEventVersions()
```

Comment:
i think we should have some logs in here for when we delete tables... probably same for when we create/delete in the other Purge method

Reply:
I've added a generic log at the start of this method, but would like to not add looks to the foreach loop - that would be chaos :D

Reply:
Can we add logic for the table operations, though? Not every loop, but any time the evaluation results in a table deletion, it would be good to log.

Reply:
I can add this, but my concern is still the same; if there are 10s of tables, we could be writing a log for each time we delete which is excessive? Maybe this is not a real concern as we purge every month anyway so we are not really dealing with any more than a table or 2?

Reply:
There should only be one table in most cases, I think? We should be purging old tables every time we create a new one, so it shouldn't be too chaotic.